### PR TITLE
Migrate from optparse to argparse

### PR DESCRIPTION
**Pull request by mandolaerik**
- Shut up linter
- Remove the --illegal-attrs flag
- Remove --illegal-attrs flag
- Remove --version, which was ignored
- Migrate from optparse to argparse

### CODE REVIEW

**Comment by syssimics** [comment]
Verification [#12224557](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=12224557): pass

**Comment by syssimics** [comment]
Verification [#12227474](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=12227474): pass

File: py/dml/dmlc.py [hunk 1]
```diff
@@ -281,102 +300,98 @@ def main(argv):
...
+        help='generate makefile dependencies')
 
     # <dt>--no-dep-phony</dt>
     # <dd>With --dep, avoid addition of a phony target for each dependency
     # other than the main file.</dd>
-    optpar.add_option(
-        '--no-dep-phony', dest = "no_dep_phony", action = 'store_true',
-        help = 'With --dep, avoid addition of a phony target for each'
+    parser.add_argument(
+        '--no-dep-phony', action='store_true',
```
Comment (lwaern-intel) [inline py/dml/dmlc.py:330]: So with no explicit `dest` the parser will parse `--no-dep-phony`, and turn it into `no_dep_phony` when genning the options? Huh.

**Review by lwaern-intel — APPROVED** [review]
Undeniably uncontroversial

File: py/dml/messages.py [hunk 2]
```diff
@@ -2392,3 +2392,7 @@ class PINT1(PortingMessage):
     types, then the value of the variable becomes 1, whereas for
     `int1` in DML 1.4 the value is -1."""
     fmt = "Change int1 to uint1"
+
+warnings = {name: cls for (name, cls) in globals().items()
```
Comment (lwaern-intel) [inline py/dml/messages.py:2396]: Just to double-check that I remember the semantics of `globals()` -- does it return the accessible vars _defined in the current module only_, or does it return accessible vars accessible at the top-level of the current module, no matter where they are defined?

Comment (mandolaerik) [reply py/dml/messages.py:2396]: Somewhere inbetween, it seems -- globals() returns a dict that includes some builtin symbols like `__name__`,  `__loader__` and `__builtins__`, but not others like `__import__`.

File: py/dml/dmlc.py (hunk 1 above)
Comment (mandolaerik) [reply py/dml/dmlc.py:330]: Yes, that's standard practice for argparse. Adn `add_argument('-f', '--foo-bar', ...)` stores it in `foo_bar`.

**Comment by syssimics** [comment]
Verification [#12247784](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=12247784): pass

**Pull request merged**